### PR TITLE
Add custom error example (showcases Result<Success, Failure>)

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR adds a new Slice example, CustomError.

Sample output:
Client
```
The greeting for '' is EmptyName { }
The greeting for 'jimmy' is Away until 1/24/2024 3:33:17 PM
The greeting for 'billy bob' is NameTooLong { MaxLength = 7 }
The greeting for 'alice' is 'Hello, alice!'
The greeting for 'bernard' is 'Hello, bernard!'
```

Server
```
Dispatching greet request { name = '' }
Dispatching greet request { name = 'jimmy' }
Dispatching greet request { name = 'billy bob' }
Dispatching greet request { name = 'alice' }
Dispatching greet request { name = 'bernard' }
```

### CODE REVIEW

File: examples/slice/CustomError/Client/Program.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+var greeter = new GreeterProxy(connection);
+
+string[] names = [ "", "jimmy", "billy bob", "alice", Environment.UserName ];
+
+foreach (string name in names)
+{
+    Console.Write($"The greeting for '{name}' is ");
+
+    // Use the Dunet-generated MatchXxx methods to process the result and the GreeterError.
+    string message = await greeter.GreetAsync(name).MatchAsync(
```
Comment (pepone) [inline examples/slice/CustomError/Client/Program.cs:18]: I think it is clear to not use the MatchAsync extension.

```
Result<string, GreeterError> result = await greeter.GreetAsync(name);
result.Match(...);
```

**Review by pepone — APPROVED** [review]

File: examples/slice/CustomError/Client/Program.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+await using var connection = new ClientConnection(new Uri("icerpc://localhost"));
+
+var greeter = new GreeterProxy(connection);
+
+string[] names = [ "", "jimmy", "billy bob", "alice", Environment.UserName ];
+
+foreach (string name in names)
+{
+    Console.Write($"The greeting for '{name}' is ");
```
Comment (externl) [inline examples/slice/CustomError/Client/Program.cs:15]: I think there should be a different message for empty and too long error cases. It's weird to see (or whatever failure.Value is for name too long):

```
The greeting for 'billy bob' is GreeterError.NameTooLong(7)
```

I also think that the getting should be wrapped in quotes like the name.

**Review by externl — APPROVED** [review]

**Review by InsertCreativityHere — APPROVED** [review]
Looks good to me!

File: examples/slice/CustomError/Client/Program.cs (hunk 2 above)
Comment (bernardnormier) [reply examples/slice/CustomError/Client/Program.cs:15]: You meant the failure?

Comment (bernardnormier) [reply examples/slice/CustomError/Client/Program.cs:15]: > I think there should be a different message for empty and too long error cases. 
The messages are different. As shown in the description, they are:
```
The greeting for '' is EmptyName { }
The greeting for 'billy bob' is NameTooLong { MaxLength = 7 }
```

File: examples/slice/CustomError/Client/Program.cs (hunk 1 above)
Comment (bernardnormier) [reply examples/slice/CustomError/Client/Program.cs]: Fixed as suggested.

**Pull request merged**